app/frontend/wallet/address-manager.js

```diff
@@ -9,21 +9,81 @@ const AddressManager = ({
   derivationScheme,
   disableCaching, // good for tests
   isChange,
+  blockchainExplorer,
 }) => {
   const state = {
     deriveAddressMemo: {},
   }
 
-  function discoverAddresses() {
+  async function discoverAddresses() {
+    switch (derivationScheme.type) {
+      case 'v1':
+        return await discoverAddressesV1()
+      case 'v2':
+        return await discoverAddressesV2()
+      default:
+        return null
```

Comment:
this is dangerous, we should throw an exception if an unknown scheme is supplied